feat: improve support for multipart requests

Dakskihedron opened the pull request:
### Description
Various additions to improve support for multipart request bodies.

### Changes
* Add `contentMediaType` property to `CodegenSchema` and update all extenders.
* Set value of `contentMediaType` if content requires metadata.
* Add helper function to retrieve the schema from the `value` property in metadata schemas.

Dakskihedron committed:
feat: set content media type on request schemas

Set the content media type on schemas representing a multipart request body.

karlvr commented on packages/types/src/types.ts:
```diff
@@ -553,6 +553,8 @@ export interface CodegenSchema extends CodegenSchemaInfo {
 	 */
 	purpose: CodegenSchemaPurpose
 
+	contentMediaType: CodegenMediaType | null
```
Let's please add docs:

```typescript
/** If this schema was created to represent a specific request content type, this property will be populated with that media type. */
```

Dakskihedron committed:
docs: add comment for contentMediaType

Dakskihedron committed:
feat: add helper to fetch metadata value schema

Add a helper that fetches the “value” schema from a metadata schema that might be an object or array schema.

karlvr commented on packages/utils/src/content.ts:
```diff
@@ -0,0 +1,17 @@
+import { CodegenSchema, isCodegenObjectLikeSchema, isCodegenArraySchema } from '@openapi-generator-plus/types'
+
+/**
+ * Return the "value" schema for a metadata schema that might be an object schema or array schema.
```
Perhaps "that might be either single-valued or an array".

karlvr commented on packages/utils/src/content.ts:
```diff
@@ -0,0 +1,17 @@
+ * @param schema 
+ * @returns 
+ */
+export function valueSchemaForMetadataSchema(schema: CodegenSchema): CodegenSchema | null {
```
Actually, I think in the core generator we'd usually return `CodegenSchema | undefined` from functions. The `null` thing is something for the object properties themselves.

Dakskihedron committed:
refactor: change null to undefined

And improve some wording.

karlvr commented on packages/utils/src/content.ts:
```diff
@@ -1,17 +1,17 @@
 /**
- * Return the "value" schema for a metadata schema that might be an object schema or array schema.
+ * Return the "value" schema for a metadata schema that might be either single-valued or an array.
  * Otherwise return null if such a schema cannot be found.
```
Need to update this comment to replace `null` with `undefined`.
